keyboards/framework/laptop16/framework.c

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022-2023 Framework Computer
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+#include "framework.h"
+
+void keyboard_post_init_kb(void) {
+    keyboard_post_init_user();
+
+    // Enable debug output
+    debug_enable   = true;
+    debug_matrix   = true;
+    debug_keyboard = true;
+}
+
+/**
+ * Hook into early keyboard initialization
+ */
+void keyboard_pre_init_kb(void) {
+    keyboard_pre_init_user();
+
+    // Mark boot as done.
+    // Before this, when holding down both alt keys QSPI_SS is pulled low to put
+    // the RP2040 in bootloader mode during reset.
+    setPinOutput(BOOT_DONE_GPIO);
+    writePinLow(BOOT_DONE_GPIO);
+
+    // TODO: Do we ever need to disable it to save power?
+    setPinOutput(MUX_ENABLE_GPIO);
+    writePinHigh(MUX_ENABLE_GPIO);
+
+    setPinInput(SLEEP_GPIO);
+}
```

Comment:
Might be simpler to move the `process_record_*` stuff here. and remove the "framework.h" include from the keymaps. Eg:
```suggestion
}

// Make sure to keep FN Lock even after reset
bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
    if (!process_record_user(keycode, record)) {
        return false;
    }
    switch (keycode) {
        case FN_LOCK:
            if (record->event.pressed) {
                if (layer_state_is(1)) {
                    set_single_persistent_default_layer(2);
                }
                if (layer_state_is(3)) {
                    set_single_persistent_default_layer(0);
                }
            }
            return false;
            break;
        default:
            break;
    }
    return true;
}

```

However, this would require either changing the keycode from `FN_LOCK` to `QK_KB` (and/or adding a define to alias it), or adding `ansi.h` in the ansi folder with just: 
```c
// Copyright 2022-2023 Framework Computer
// SPDX-License-Identifier: GPL-2.0-or-later

#pragma once

#include "framework.h"
```

I'm not sure this is the best way to handle this, but would move the custom code out of the keymaps, and remove the need for the framework.h include in each of the keymaps.

Reply:
Isn't it more of layout code though?
If people want to make their own layouts and have FN lock on a different layer, they'd need to change this code.

Unfortunate to duplicate it, but I guess that's unavoidable, since the layout itself is also basically duplicated.